docs/src/rules/no-invalid-this.md

```diff
@@ -284,6 +284,51 @@ obj.Foo = function Foo() {
 
 :::
 
+This rule additionally supports TypeScript type syntax.
```

Comment:
Let's also add a small note similar to this

<img width="1016" alt="Screenshot 2025-03-21 at 3 32 52 PM" src="https://github.com/user-attachments/assets/9b1f15fb-0fb8-4565-935d-d321d2116a68" />